### PR TITLE
Updated logging for frontstage

### PR DESCRIPTION
**Pull request by emilioward**
**Context of this PR:**

This PR updates the logging on the front stage (Trello card - https://trello.com/c/ClGvqx2W)

**How to review:**

Look through changed log code then run through the application to see if the correct logs appear in console

### CODE REVIEW

File: frontstage/views/register.py [hunk 1]
```diff
@@ -27,16 +27,17 @@ def validate_enrolment_code(enrolment_code):
     # Build the URL
     # url = app.config['API_GATEWAY_IAC_URL'] + '{}'.format(enrolment_code)
     url = app.config['RM_IAC_GET'].format(app.config['RM_IAC_SERVICE'], enrolment_code)
-    logger.debug('Validate IAC URL is: {}'.format(url))
+    logger.debug('"event" : "Logging IAC URL", "URL": "{}"'.format(url))
```
Comment (andrewmil) [inline frontstage/views/register.py]: I think splunk will be fine with this, though to be exact I think it's better to be `event: "Logging IAC URL"` without the space before : or the quotes around event. We'll have to ask splunk guys about exact formatting

File: frontstage/views/register.py [hunk 2]
```diff
@@ -27,16 +27,17 @@ def validate_enrolment_code(enrolment_code):
...
     result = requests.get(url, verify=False)
-    logger.debug('Result => {} {} : {}'.format(result.status_code, result.reason, result.text))
+    logger.debug('"event" : "Validate enrolment code", "result status code": "{}:, "result reason" : "{}", '
+                 '"result text" : "{}"'.format(result.status_code, result.reason, result.text))
 
     if result.status_code == 200 and json.loads(result.text)['active']:
         case_id = json.loads(result.text)['caseId']
     elif result.status_code == 404:
-        logger.error("Iac code not found code: {}".format(enrolment_code))
+        logger.error('"event" : "error", "Iac code not found: code" : "{}"'.format(enrolment_code))
```
Comment (andrewmil) [inline frontstage/views/register.py]: This line doesnt make much sense, 5 things so it's not in key-value pairs

File: frontstage/views/register.py [hunk 3]
```diff
@@ -77,7 +78,7 @@ def register():
 
             return redirect(url_for('register_bp.register_confirm_organisation_survey', enrolment_code=coded_token))
         else:
-            logger.info('Invalid IAC code: {}'.format(enrolment_code))
+            logger.info('"event" : "error", "Iac code invalid" : "{}"'.format(enrolment_code))
```
Comment (andrewmil) [inline frontstage/views/register.py]: `logger.error('"event" : "Iac code invalid", "Iac code" : "{}"'.format(enrolment_code))`

File: frontstage/views/register.py [hunk 4]
```diff
@@ -104,22 +105,22 @@ def register_confirm_organisation_survey():
     if encrypted_enrolment_code:
         decrypted_enrolment_code = ons_env.crypt.decrypt(encrypted_enrolment_code.encode()).decode()
     else:
-        logger.error('Confirm organisation screen - Enrolment code not specified')
+        logger.error('"event" : "error", "message" : "Confirm organisation screen - Enrolment code not specified"')
```
Comment (andrewmil) [inline frontstage/views/register.py]: The event should be your message here. The .error let's you know that it's an error

File: frontstage/views/register.py [hunk 5]
```diff
@@ -211,15 +214,8 @@ def register_enter_your_details():
         last_name = request.form.get('last_name')
         email_address = request.form.get('email_address')
         password = request.form.get('password')
-        password_confirm = request.form.get('password_confirm')
         phone_number = request.form.get('phone_number')
 
-        logger.debug("User name is: {} {}".format(first_name, last_name))
-        logger.debug("Email is: {}".format(email_address))
-        logger.debug("password is: {}".format(password))
```
Comment (andrewmil) [inline frontstage/views/register.py]: Can't believe this was here 😂

File: frontstage/views/register.py [hunk 6]
```diff
@@ -229,7 +225,7 @@ def register_enter_your_details():
         # this deals with the transactions between the OAuth2 server
         oauth = OAuth2Session(client=client)
         token_url = app.config['ONS_OAUTH_PROTOCOL'] + app.config['ONS_OAUTH_SERVER'] + app.config['ONS_TOKEN_ENDPOINT']
-        logger.debug("Our Token Endpoint is: {}".format(token_url))
+        logger.debug('"event" : "token url", "Our Token Endpoint" : "{}"'.format(token_url))
```
Comment (andrewmil) [inline frontstage/views/register.py]: `'event: "fetching oauth token", url: "{}"'.format(token_url)`

File: frontstage/views/register.py [hunk 7]
```diff
@@ -256,14 +252,14 @@ def register_enter_your_details():
 
         except JWTError:
             # TODO Provide proper logging
-            logger.warning('JWT scope could not be validated.')
-            return abort(500, '{"message":"There was a problem with the Authentication service please contact a member of the ONS staff"}')
+            logger.warning('"event" : "JWT scope could not be validated."')
+            return abort(500, '{"message" : "There was a problem with the Authentication service please contact a member of the ONS staff"}')
```
Comment (andrewmil) [inline frontstage/views/register.py]: event instead of message

**Comment by dcdarrell9** [comment]
There are a few parts where  `ons_env.logger` is used and that's different logging, not sure about how that works though.

File: frontstage/views/register.py (hunk 5 above)
Comment (dcdarrell9) [inline frontstage/views/register.py]: 👍 💯 🥇

File: frontstage/views/register.py [hunk 8]
```diff
@@ -57,7 +58,7 @@ def register():
     if request.method == 'POST' and form.validate():
 
         enrolment_code = request.form.get('enrolment_code')
-        logger.debug("Enrolment code is: {}".format(enrolment_code))
+        logger.debug('"event" : "Logging enrolment code", "Enrolment code" : "{}"'.format(enrolment_code))
```
Comment (andrewmil) [inline frontstage/views/register.py]: enrolment code and iac code are the same thing so should pick one

File: frontstage/views/register.py [hunk 9]
```diff
@@ -198,7 +201,7 @@ def register_enter_your_details():
 
     # Ensure we have got a valid enrolment code, otherwise go to the sign in page
     if not decrypted_enrolment_code or not validate_enrolment_code(decrypted_enrolment_code):
-        logger.error('Enter Account Details page - invalid enrolment code: ' + str(decrypted_enrolment_code))
+        logger.error('"event" : "invalid code", "error" : "Enter Account Details page - invalid enrolment code: "' + str(decrypted_enrolment_code))
```
Comment (andrewmil) [inline frontstage/views/register.py]: Don't need event and error here, 
`'"event": "Invalid enrolment code", "Iac_code": "{}"'.format(str(decrypted_enrolment_code))

File: frontstage/views/register.py [hunk 10]
```diff
@@ -256,14 +249,14 @@ def register_enter_your_details():
...
             # TODO Provide proper logging
-            logger.warning('JWT scope could not be validated.')
-            return abort(500, '{"message":"There was a problem with the Authentication service please contact a member of the ONS staff"}')
+            logger.warning('"event" : "JWT scope could not be validated."')
+            return abort(500, '{"event" : "There was a problem with the Authentication service please contact a member of the ONS staff"}')
 
         except MissingTokenError as e:
-            logger.warning("Missing token error, error is: {}".format(e))
-            logger.warning("Failed validation")
+            logger.warning('"event" : "missing token", "Missing token, error is" : "{}"'.format(e))
```
Comment (andrewmil) [inline frontstage/views/register.py]: logger.warning('"event" : "missing token", "Missing token, error is" : "{}"'.format(e))
this doesn't follow the key value pair pattern

File: frontstage/views/register.py [hunk 11]
```diff
@@ -281,10 +274,10 @@ def register_enter_your_details():
 
         # party_service_url = app.config['API_GATEWAY_PARTY_URL'] + 'respondents'
         party_service_url = app.config['RAS_PARTY_POST_RESPONDENTS'].format(app.config['RAS_PARTY_SERVICE'])
-        logger.debug("Party service URL is: {}".format(party_service_url))
+        logger.debug('"event" : "get party url", "Party service URL is" : "{}"'.format(party_service_url))
```
Comment (andrewmil) [inline frontstage/views/register.py]: `URL:` not `Party service URL is:`

File: frontstage/views/register.py [hunk 12]
```diff
@@ -281,10 +274,10 @@ def register_enter_your_details():
...
 
         result = requests.post(party_service_url, headers=headers, data=json.dumps(registration_data))
-        logger.debug("Response from party service is: {}".format(result.content))
+        logger.debug('"event" : "party url", "Response from party service is" : "{}"'.format(result.content))
```
Comment (andrewmil) [inline frontstage/views/register.py]: same as above here. I only commented the top stuff before so you might need to go through these files too to make it consistent

**Pull request merged**